src/components/editor/failureMode/FailureModesList.tsx

```diff
@@ -67,7 +69,12 @@ const FailureModesList = ({   label,
                 renderValue={(selected: any[]) => formatOutput(selected.filter(v => v).map(value => value.name).join(", "), 65)}
             >
 
-                {(Array.from(allFailureModes.values())).map((failureMode) =>
+                {
+                    (Array.from(allFailureModes.values()))
+                        .filter(fm => {
+                            if (allowCauses) return true;
+                            else return !allowCauses && fm.failureModeType === FailureModeType.FailureMode;
+                        }).map((failureMode) =>
```

Comment:
This code in the return statement makes understanding what is rendered hard to read. May be it will be better to calculate the list of FMs and store it in a **const** and use here.

Reply:
It makes sense.